content/docs/refs-and-the-dom.md

```diff
@@ -66,24 +105,26 @@ class CustomTextInput extends React.Component {
 }
 ```
 
-React will call the `ref` callback with the DOM element when the component mounts, and call it with `null` when it unmounts. `ref` callbacks are invoked before `componentDidMount` or `componentDidUpdate` lifecycle hooks.
-
-Using the `ref` callback just to set a property on the class is a common pattern for accessing DOM elements. The preferred way is to set the property in the `ref` callback like in the above example. There is even a shorter way to write it: `ref={input => this.textInput = input}`. 
+React will assign the `value` property with the DOM element when the component mounts, and assign it back to `null` when it unmounts. `ref` updates happen before `componentDidMount` or `componentDidUpdate` lifecycle hooks.
 
-### Adding a Ref to a Class Component
+#### Adding a Ref to a Class Component
 
-When the `ref` attribute is used on a custom component declared as a class, the `ref` callback receives the mounted instance of the component as its argument. For example, if we wanted to wrap the `CustomTextInput` above to simulate it being clicked immediately after mounting:
+If we wanted to wrap the `CustomTextInput` above to simulate it being clicked immediately after mounting, we could use a ref to get access to the custom input and call its `focusTextInput` method manually:
 
-```javascript{3,9}
+```javascript{4,7,12}
```

Comment:
The line-highlights are still off for this example. You can see it here:
https://deploy-preview-637--reactjs.netlify.com/docs/refs-and-the-dom.html#adding-a-ref-to-a-class-component

Should be {4,8,13} rather than {4,7,12}.

This (along with Prettier formatting) is why [I suggested moving these examples](https://github.com/reactjs/reactjs.org/pull/637/files#r170019692) to the examples folder. It's easier to manage highlights inline rather than at the top of the code snippet. 😄

Reply:
The line highlights are still off for this example...

<img width="438" alt="screen shot 2018-03-07 at 10 35 05 am" src="https://user-images.githubusercontent.com/29597/37110982-56c96440-21f3-11e8-9982-a0f195f0dd02.png">